### PR TITLE
simplify c api

### PR DESCRIPTION
**Pull request by rageagainsthepc**
Another box around the box could have spared us a lot of work. ;)

### CODE REVIEW

File: src/driver/hyperv.rs [hunk 1]
```diff
@@ -255,12 +255,6 @@ impl HyperV {
     }
 }
 
-impl Introspectable for HyperV {
```
Comment (Wenzel) [inline src/driver/hyperv.rs:258]: I think you need to leave an empty Impl for Hyper-V, where all methods returns unimplemented(), otherwise the CI fails

Comment (rageagainsthepc) [reply src/driver/hyperv.rs:258]: Actually, this wasn't intentional. I did not want to delete the whole impl block. :scream:

File: Cargo.toml [hunk 2]
```diff
@@ -24,7 +24,7 @@ libc = { version = "0.2.58", optional = true }
 xenctrl = { git = "https://github.com/Wenzel/xenctrl", optional = true }
 xenstore = { git = "https://github.com/Wenzel/xenstore", optional = true }
 xenforeignmemory = { git = "https://github.com/Wenzel/xenforeignmemory", optional = true }
-kvmi = { git = "https://github.com/Wenzel/kvmi", optional = true }
+kvmi = { git = "https://github.com/Wenzel/kvmi", rev = "b9d8df6730b46f6a00c9ba6e1adcdd5434f613d7", optional = true }
```
Comment (Wenzel) [inline Cargo.toml:27]: What happens if we use master branch ?
I don't want to commit a specific revision here.
Also, we should switch to published crates.

Comment (rageagainsthepc) [reply Cargo.toml:27]: Libmicrovmi is currently broken on master after the last kvmi merge. Why not pin a commit? It is functionally equivalent to pinning a version and we don't have published crates right now.

Comment (rageagainsthepc) [reply Cargo.toml:27]: I consider this a temporary solution until we either merge a PR that is compatible with the latest changes in kvmi or start publishing crates on crates.io

Comment (Wenzel) [reply Cargo.toml:27]: now it makes sense

**Comment by Wenzel** [comment]
Can I merge this one ?

**Comment by rageagainsthepc** [comment]
Yes, go ahead ;)

**Pull request merged**